keylime/ca_util.py

```diff
@@ -63,23 +63,31 @@ def load_cert_by_path(cert_path: str) -> Certificate:
     return cert
 
 
-def setpassword(pw: Optional[str]) -> None:
+def setpassword(key_store_pw: Optional[str], key_usage: Optional[str] = "server") -> None:
     global global_password
-    if not pw:
-        pw = getpass.getpass("Please enter the password to decrypt your keystore: ")
+    if not key_store_pw and not global_password:
+        key_store_pw = config.get("ca", "password", fallback="default")
+        if key_store_pw == "default":
+            if key_usage == "server":
+                logger.warning("Using 'default' password option from CA configuration file")
```

Comment:
Somehow `setpasword` is not anymore the correct function. Now is acting as "set password from the CLI", "ask password" and "read password from config file". I think that this should be splited in three and used when makes sense, and avoid the new `key_usage` pollution in the different signatures.

Reply:
Uhmm one option can be to have `set_password(key_store_pw: str)`, `read_password() -> str` and `ask_password() -> str`. The invariant now can be that `read_private(warn: bool = False) -> Tuple[Dict[str, Any], str]` expect that the `global_password` is always set (we can put an assert there and exit if this is not the case) but will never ask nor read the password as was done before.

Before using any `ca_util` function, the user will be sure that the password is set, calling always `set_password()` with the output of the read or ask version, depending if the entry point is from a cli or not.  That way we can remove `key_usage` from all the signatures.

Reply:
Hmmmm... I think I get it, but let me restate it for clarity: instead of letting the internal logic in `setpassword` to decide when it is acceptable to read the password from the configuration file, we simply move it to this decision to individual invocation of `setpassword`, replacing it with `read_password` and `ask_password` when needed?

Reply:
I think I can see the logic of what you're proposing, given that the server-side components (i.e., `verifier`) will never actually prompt a user for a password. Let me rework this a bit more then.

Reply:
Yes, thank you for using better wording!